services/minespace-web/src/components/Forms/tailing/tailingsStorageFacility/TailingsSummaryForm.js

```diff
@@ -0,0 +1,219 @@
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { withRouter } from "react-router-dom";
+import { flattenObject } from "@common/utils/helpers";
+import { bindActionCreators, compose } from "redux";
+import { isNil } from "lodash";
+import {
+  arrayPush,
+  change,
+  formValueSelector,
+  getFormSyncErrors,
+  getFormValues,
+  reduxForm,
+} from "redux-form";
+import { LeftOutlined, RightOutlined } from "@ant-design/icons";
+import { Form } from "@ant-design/compatible";
+import "@ant-design/compatible/assets/index.css";
+import { Button, Col, Popconfirm, Row } from "antd";
+import * as FORM from "@/constants/forms";
+import CustomPropTypes from "@/customPropTypes";
+import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
+import BasicInformation from "@/components/Forms/tailing/tailingsStorageFacility/BasicInformation";
+import EngineerOfRecord from "@/components/Forms/tailing/tailingsStorageFacility/EngineerOfRecord";
+import LinkButton from "@/components/common/LinkButton";
+import { AuthorizationGuard } from "@/HOC/AuthorizationGuard";
+import * as Permission from "@/constants/permissions";
+
+const propTypes = {
+  initialValues: CustomPropTypes.projectSummary.isRequired,
+  match: PropTypes.shape({
+    params: PropTypes.shape({
+      tab: PropTypes.string,
+    }),
+  }).isRequired,
+  mines: CustomPropTypes.mine.isRequired,
+  submitting: PropTypes.bool.isRequired,
+  change: PropTypes.func.isRequired,
+  formErrors: PropTypes.objectOf(PropTypes.string),
+  handleTabChange: PropTypes.func.isRequired,
+  formValues: PropTypes.objectOf(PropTypes.string),
+  handleSaveData: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  formValues: {},
+  formErrors: {},
+};
+
+const tabs = [
+  "basic-information",
+  "engineer-of-record",
+  "qualified-person",
+  "registry-document",
+  "reports",
+  "summary",
+];
+
+export const TailingsSummaryForm = (props) => {
+  const [tabIndex, setTabIndex] = useState(0);
+  const {
+    initialValues,
+    match,
+    submitting,
+    handleTabChange,
+    formErrors,
+    handleSaveData,
+    formValues,
+  } = props;
+
+  useEffect(() => {
+    setTabIndex(tabs.indexOf(match.params.tab));
+  }, []);
+
+  useEffect(() => {
+    if (tabIndex !== tabs.indexOf(match.params.tab)) {
+      setTabIndex(tabs.indexOf(match.params.tab));
+    }
+  }, [match.params.tab]);
+
+  const renderTabComponent = (tab) =>
+    ({
+      "basic-information": <BasicInformation initialValues={initialValues} />,
+      "engineer-of-record": <EngineerOfRecord initialValues={initialValues} />,
+      "qualified-person": <div />,
+      "registry-document": <div />,
+      reports: <div />,
+      summary: <div />,
+    }[tab]);
+  const isFirst = tabIndex === 0;
+  const isLast = tabs.length - 1 === tabIndex;
+  const errors = Object.keys(flattenObject(formErrors));
+  const disabledButton = errors.length > 0;
+  return (
```

Comment:
I see that this is very similar to the project summary do you think it would be a lot of effort to make this into a reusable component? E.g. you would be able to pass along the tabs you want, and it internally manages which tab is active, whether or not to show the back button etc. I get the feeling that we'll have more pages like these popping up in MineSpace going forwards

Reply:
Good thought.  It could certainly be done.  I'd say it would be a moderate effort to set that up, but it could definitely be worth it.

Reply:
Sounds good! Maybe let's check in with Roop to see if he imagines it being used elsewhere, and we can do the refactor with one of the stories this sprint